taskvine/src/manager/vine_current_transfers.c

```diff
@@ -135,6 +135,22 @@ int vine_current_transfers_dest_in_use(struct vine_manager *q, struct vine_worke
 	return c;
 }
 
+// return true if worker w is currently a receiving file cachename, else false
+int vine_current_transfers_file_worker_recv(struct vine_manager *q, struct vine_worker_info *w, const char *cachename)
+{
+
+	char *id;
+	struct vine_transfer_pair *t;
+
+	HASH_TABLE_ITERATE(q->current_transfer_table, id, t)
```

Comment:
This is inefficient. Instead check if worker is in the value of the table m->file_worker_table. See `vine_file_replica_table_insert`.